include/zephyr/drivers/key.h

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2022 tangchunhui@coros.com
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
```

Comment:
No new APIs without documentation in Sphinx. You need to update doc/develop/api/overview.rst for this API, and add a page documenting this new API in .rst just like the other APIs mentioned in overview.rst.